### PR TITLE
[RFR] Feature/77 notes comments

### PR DESCRIPTION
**Pull request by enginehouse**
This is is doubtless going to be handed back as it is mid design - also the design for file upload needs to be reigned in to align with the Gov Uk guidelines

### CODE REVIEW

**Comment by enginehouse** [comment]
@Koded - is adding a deadline to the actual comment part of this sprint? I have it as an option but I haven't made it carry it through to the comment and format it (jQuery is nasty for templating : ))
If not then please can I have a review on this? I need to set webpack up to bundle javascripts and allow me to require the plugins and make proper controllers but I haven't done that yet

**Comment by Koded** [comment]
Hi @enginehouse - the deadline functionality is part of story 90, so no, you don't need to complete it as part of 77. 

So i'll review...

File: app/shared/components/FileUpload/component.jsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+const FileUpload = props => {
+  let classes = classNames('form-group form-group-file', props.className, props.modifiers)
+  return (
+    <div className={classes} data-file>
+      {props.shimmed && <span className='form-control__text' data-file-text></span>}
+      <label className='form-label' htmlFor={'file-' + props.id}>
+        {props.label}
+      </label>
+      <input className='form-control' type='file' id={'file-' + props.id} data-file-input/>
```
Comment (Koded) [inline app/shared/components/FileUpload/component.jsx:13]: Inline with the guidance (https://govuk-elements.herokuapp.com/form-elements/#form-file-upload) we really need to use native upload controls.

File: app/server/api/static/application-overview-frequency-comments.yml [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  commentBlock:
+    limit: 3
+    deadline:
```
Comment (Koded) [inline app/server/api/static/application-overview-frequency-comments.yml]: I know they are both being worked on, but to make sure we only QA 77, and that we can complete 77, comment this out until we are ready to complete story 90. 

(also the other block)

File: app/shared/components/FileUpload/component.jsx (hunk 1 above)
Comment (Koded) [reply app/shared/components/FileUpload/component.jsx:13]: (know you made a note of this at the top of the PR)

File: app/shared/components/Scripts/component.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+
+const Scripts = () => {
+  return (
+    <div>
```
Comment (Koded) [inline app/shared/components/Scripts/component.jsx]: React 16 allows for rendering without wrapping div, something like:

```
const ScriptItems = () => ([
  <script src='/ui/js/vendor/jquery.3.3.1.min.js'/>
  <script src='/ui/js/plugins/file-upload-shim.js'/>
]);
```

Comment (enginehouse) [reply app/shared/components/Scripts/component.jsx]: Ahh fabulous - I didn't even realise we had upgraded it : ) - I'll chop that extraneous div out (there may be other instances of this)

Comment (enginehouse) [reply app/shared/components/Scripts/component.jsx]: Addressed this - removed now

File: app/server/api/static/application-overview-frequency-comments.yml (hunk 2 above)
Comment (enginehouse) [reply app/server/api/static/application-overview-frequency-comments.yml]: Removed the deadline component of this

File: app/shared/components/FileUpload/component.jsx (hunk 1 above)
Comment (enginehouse) [reply app/shared/components/FileUpload/component.jsx:13]: Agreed - it still requires the shim to return the value of the file input but the control itself is the native one now.

**Comment by enginehouse** [comment]
Hello @Koded - have tweaked this request with your updates.

**Review by Koded — APPROVED** [review]

**Pull request merged**